### DISCUSSION TITLE
Relational Repeater Interface

### DISCUSSION THREAD

**Discussion by claidler** [directus/directus#3962] [Feature Requests]
Firstly - great work on this CMS. It's really nice. 

I'm not sure if I'm missing something, but I can't see any way of getting assets, images or videos into repeater fields?

Am I missing something, or is this something you're planning to add in at some point?

Thanks.

**Comment by benhaynes** [comment]
Thanks! The current repeater stores in json, so it can't have relational info. We'll be making a relational repeater that can handle that data, eg: images

**Reply by claidler** [reply]
Great! Can't wait. Thank you

**Reply by benfeather** [reply]
+1 great idea!

**Reply by sam-page-dc** [reply]
+1

**Reply by vtisnado** [reply]
Do you have any estimate date for this feature?

**Reply by rijkvanzanten** [reply]
@vtisnado it's the one to many interface 👍🏻

**Reply by benhaynes** [reply]
I think we should add an option on the repeater and O2M interface that allows the fields to be displayed on the form, instead of in a drawer. I know we've gone back and forth on this quite a bit, mostly because of nested grouping, but there should be a solution for the easiest implementations.

**Reply by faridsaud** [reply]
+1

**Reply by devjabeer** [reply]
A dedicated repeater type would be great.

**Reply by JLKM** [reply]
An O2M interface on a form showing related records (with the possibility to add more) is a big wish of mine. Could work like as an embedded list with its own template - filtered on the related parent record. It's just about the only thing I miss on this absolutely wonderful project.

**Reply by JLKM** [reply]
Sorry. My bad. Directus 9 is even more than capable of what I wished for. I got it working by using the "Create Field in Advanced Mode" and create the relations from the M2O-side - including the corresponding O2M-field. It works like a charm. Really impressive with all that templating + fetching fields across tables. Pretty mindblowing, what has to be going on behind the scenes here. Nothing now prevents me from going ahead with my project. Thanks again for this awesome project.

**Comment by TomS-** [comment]
+1

I use repeaters a lot, mostly for blogs as you can build out the content better or something like this https://statamic.dev/fieldtypes/bard?ref=docs.statamic.com would be a DREAM.

**Comment by stouch** [comment]
For those who would prefer to still use json repeater, with a File Selector on uuid type field, I may (not sure of "when") going to develop an extension  : https://github.com/directus/directus/discussions/3001#discussioncomment-1652323 

If you have thought about something like that , do not hesitate to give feedback of the approach

**Comment by stouch** [comment]
Ok so I've made up something that is working : https://github.com/stouch/directus-interface-uuid-file-image

Do not hesitate to contribute.

Have a nice week !

**Comment by rijkvanzanten** [comment]
Closing this as a duplicate of https://github.com/directus/directus/discussions/3884 🙂
